daemon/graphdriver/overlay2/overlay.go

```diff
@@ -266,7 +266,7 @@ func supportsOverlay() error {
 
 func useNaiveDiff(home string) bool {
 	useNaiveDiffLock.Do(func() {
-		if err := hasOpaqueCopyUpBug(home); err != nil {
+		if err := hasOpaqueCopyUpBugOrEnableRedirectDir(home); err != nil {
```

Comment:
So this is called on every invocation of Diff() and other places. I am wondering why it was not done once during initialization time and save the state in some boolean variable and use that for rest of the operations.